app/controllers/case_logs_controller.rb

```diff
@@ -9,21 +9,18 @@ class CaseLogsController < ApplicationController
   before_action :find_resource, except: %i[create index edit]
 
   def index
-    set_session_filters
-
-    all_logs = current_user.case_logs
-    unpaginated_filtered_logs = filtered_case_logs(filtered_collection(all_logs, search_term))
+    filters = set_updated_session_filters
```

Comment:
Personal preference but I think nouns over verbs for methods seems more idiomatic (it doesn't seem like this one has side effects).

Reply:
I struggled with naming this; it has side effects (it sets the filters on the session for reading back next time), but we also need access to those filters for the E-mail job.

My immediate inclination was splitting it into two parts (the function to get the filters, the function to set them on the session), but I think in practice any time we're getting them we'll also want to update the session. I was avoiding the need to always call both together.

Welcome to suggestions on naming or organisation on this, I'm certainly not entirely happy with it as-is 😄

Reply:
Yes, I think it would be better to split out into two parts. The first sets the filters to the session and the second is a getter for either the raw filters or the filtered case logs.

```ruby
  before_action :set_session_filters # Or, keep inline in actions.
  before_action :find_session_filters

  private

  # Inline just for example
  def set_session_filters
    ...
  end

  def find_session_filters
    @session_filters ||= load_session_filters(session[:case_logs_filters]) # extract this method from set_session_filters
  end
```

Reply:
Don't we need an unconditional assignment to `@session_filters` (since we're looking to update it?)